libminifi/src/c2/C2Agent.cpp

```diff
@@ -596,6 +619,29 @@ void C2Agent::handle_describe(const C2ContentResponse &resp) {
       }
       enqueue_c2_response(std::move(response));
     }
+  } else if (resp.name == "corecomponentstate") {
```

Comment:
This has only been tested manually. Once https://github.com/apache/nifi-minifi-cpp/pull/743 is done and merged, I can probably write reasonable tests for it.

Reply:
Added `C2DescribeCoreComponentStateTest` for this.